Hddm random access support

rjones30 opened the pull request:
A set of coordinated changes to the hddm tools and the underlying xstream library that introduce support for random access to hddm streams. Prior to this, you could skip forward by any number of events, but this required reading a few bytes per record from every event in between. For compressed streams, this was quite heavy because the full work of decompression had to be performed for the events you wan to skip over. With this new feature, on an early pass through the data you can save a position marker (128-bit token) for any event you want to include in your list, and then come back directly to it by a call to setPosition() command on the open stream. Both forward and backward motion is supported on any hddm stream.

rjones30 committed:
Merge branch 'master' of github.com:JeffersonLab/sim-recon

rjones30 committed:
Merge branch 'master' of github.com:JeffersonLab/sim-recon

rjones30 committed:
Merge branch 'master' of github.com:JeffersonLab/sim-recon

rjones30 committed:
* gustep.F [rtj]
   - add define for BACKGROUND_PROFILING special-purpose build,
     but comment it out for correct behavior the default build.

rjones30 committed:
* hddm-cpp.cpp [rtj]
   - add new features to enable random access to hddm streams. This
     is presently working for uncompressed streams and for those
     compressed with the bz2 library, but not with the zlib library.
     I will work on getting random access to zlib streams working
     later, but I want to get this new feature out in time for the
     upcoming REST production.
* xml-hddm.cpp [rtj]
   - the old code would generate a collision if two hddm templates
     had different values for a string field. This is not a collision,
     so I corrected the code to tolerate this mismatch.

rjones30 committed:
* xstream/include/xstream/bz.h,z.h [rtj]
   - add features to the compressed streambuf objects in xstream
     so that the stream can be repositioned to arbitrary places in
     the file without reading the entire stream in between.
* xstream/src/bz.cpp,z.cpp [rtj]
   - implementation code to support the above.

gluex commented:
Build status for this pull request: FAILURE

 Build log: [/work/halld/pull_request_test/sim-recon^hddm_random_access_support/make_hddm_random_access_support.log](https://halldweb.jlab.org/pull_request_test/sim-recon^hddm_random_access_support/make_hddm_random_access_support.log)
 Build report: [/work/halld/pull_request_test/sim-recon^hddm_random_access_support/report_hddm_random_access_support.txt](https://halldweb.jlab.org/pull_request_test/sim-recon^hddm_random_access_support/report_hddm_random_access_support.txt)
 Location of build: /work/halld/pull_request_test/sim-recon^hddm_random_access_support

rjones30 committed:
* extern/xstream/src/bz.cpp [rtj]
   - qualify char constants as unsigned to get rid of compiler warnings

* programs/Simulation/HDGeant/gustep.F [rtj]
   - change C-stype comments // to fortran column-1 C comments.

gluex commented:
Build status for this pull request: FAILURE

 Build log: [/work/halld/pull_request_test/sim-recon^hddm_random_access_support/make_hddm_random_access_support.log](https://halldweb.jlab.org/pull_request_test/sim-recon^hddm_random_access_support/make_hddm_random_access_support.log)
 Build report: [/work/halld/pull_request_test/sim-recon^hddm_random_access_support/report_hddm_random_access_support.txt](https://halldweb.jlab.org/pull_request_test/sim-recon^hddm_random_access_support/report_hddm_random_access_support.txt)
 Location of build: /work/halld/pull_request_test/sim-recon^hddm_random_access_support

rjones30 committed:
* external/xstream/src/bz.cpp [rtj]
   - every new release of the c++ standards gets progressively more
     fussy about char type conversions. If I write 0xff for a char
     value then is it not obvious what I want stored in that byte?

rjones30 committed:
* external/xstream/src/bz.cpp [rtj]
   - more fussing with casting char to the precise type for event context

gluex commented:
Build status for this pull request: FAILURE

 Build log: [/work/halld/pull_request_test/sim-recon^hddm_random_access_support/make_hddm_random_access_support.log](https://halldweb.jlab.org/pull_request_test/sim-recon^hddm_random_access_support/make_hddm_random_access_support.log)
 Build report: [/work/halld/pull_request_test/sim-recon^hddm_random_access_support/report_hddm_random_access_support.txt](https://halldweb.jlab.org/pull_request_test/sim-recon^hddm_random_access_support/report_hddm_random_access_support.txt)
 Location of build: /work/halld/pull_request_test/sim-recon^hddm_random_access_support

gluex commented:
Test status for this pull request: SUCCESS

  Summary: [/work/halld/pull_request_test/sim-recon^hddm_random_access_support/tests/summary.txt](https://halldweb.jlab.org/pull_request_test/sim-recon^hddm_random_access_support/tests-summary.txt)
 Logs: [/work/halld/pull_request_test/sim-recon^hddm_random_access_support/tests/log](https://halldweb.jlab.org/pull_request_test/sim-recon^hddm_random_access_support/tests-logs)

 Build log: [/work/halld/pull_request_test/sim-recon^hddm_random_access_support/make_hddm_random_access_support.log](https://halldweb.jlab.org/pull_request_test/sim-recon^hddm_random_access_support/make_hddm_random_access_support.log)
 Build report: [/work/halld/pull_request_test/sim-recon^hddm_random_access_support/report_hddm_random_access_support.txt](https://halldweb.jlab.org/pull_request_test/sim-recon^hddm_random_access_support/report_hddm_random_access_support.txt)
 Location of build: /work/halld/pull_request_test/sim-recon^hddm_random_access_support
